router/core/factoryresolver.go

```diff
@@ -447,6 +450,18 @@ func (l *Loader) Load(engineConfig *nodev1.EngineConfiguration, subgraphs []*nod
 	return &outConfig, nil
 }
 
+func (l *Loader) subgraphName(subgraphs []*nodev1.Subgraph, dataSourceID string) string {
+	i := slices.IndexFunc(subgraphs, func(s *nodev1.Subgraph) bool {
+		return s.Id == dataSourceID
+	})
+
+	if i != -1 {
+		return subgraphs[i].Name
+	}
+
+	return dataSourceID
```

Comment:
This should never happen. We should return empty string here to indicate that something is wrong.